src/google-maps/map-rectangle/map-rectangle.ts

```diff
@@ -0,0 +1,223 @@
+   */
+  @Output()
+  rectangleDrag: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('drag');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.dragend
+   */
+  @Output()
+  rectangleDragend: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('dragend');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.dragstart
+   */
+  @Output()
+  rectangleDragstart: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('dragstart');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mousedown
+   */
+  @Output()
+  rectangleMousedown: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mousedown');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mousemove
+   */
+  @Output()
+  rectangleMousemove: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mousemove');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mouseout
+   */
+  @Output()
+  rectangleMouseout: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mouseout');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mouseover
+   */
+  @Output()
+  rectangleMouseover: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mouseover');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.mouseup
+   */
+  @Output()
+  rectangleMouseup: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('mouseup');
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Rectangle.rightclick
+   */
+  @Output()
+  rectangleRightclick: Observable<google.maps.MouseEvent> =
+      this._eventManager.getLazyEmitter<google.maps.MouseEvent>('rightclick');
+
+  constructor(private readonly _map: GoogleMap) {}
+
+  ngOnInit() {
+    const combinedOptionsChanges = this._combineOptions();
+
+    combinedOptionsChanges.pipe(take(1)).subscribe(options => {
+      this._rectangle = new google.maps.Rectangle(options);
+      this._rectangle.setMap(this._map._googleMap);
+      this._eventManager.setTarget(this._rectangle);
+    });
+
+    this._watchForOptionsChanges();
+    this._watchForBoundsChanges();
+  }
+
+  ngOnDestroy() {
+    this._eventManager.destroy();
+    this._destroyed.next();
+    this._destroyed.complete();
+    for (let listener of this._listeners) {
```

Comment:
I don't think anything is being added to the `_listeners`.

Reply:
Done.